src/ui/mouse_hover.rs

```diff
@@ -0,0 +1,25 @@
+use bevy::prelude::*;
+
+use super::{BUTTON_HOVER, BUTTON_NORMAL, BUTTON_PRESSED};
+
+pub(super) fn plugin(app: &mut App) {
+    app.add_systems(Update, update);
+}
+
+type InteractiveFilter = (Changed<Interaction>, With<Button>);
```

Comment:
I personally wouldn't split this out from where it's used, since it's only used in the one place. But not a big deal.

(I do see another place in code with `(Changed<Interaction>, With<Button>)`, but not `InteractiveFilter`)

Reply:
Trying to please clippy, should we just disable warnings per case?

Reply:
What's the warning? I agree this should not be split out.

Reply:
Complex type definition

Reply:
Not getting the warning now, not sure why it's the case but I'll take it